### ISSUE TITLE
Support Non-Transparent-Framing for syslog over TCP

### ISSUE THREAD

**Issue by vlasamik** [pivotal-cf/fluent-bit-out-syslog#19]
Hi,
I have a setup where I directed pks cluster logs to fluentd aggregator.
The sink is having parameters like:

Spec:
  Enable _ Tls:  false
  Host:          10.x.y.z
  Port:           12345
  Type:          syslog

In the aggregator side I'm using fluentd's standard input parser:
<source>
    @type syslog
    @id   psksyslog_source
    protocol_type tcp
    port 12345
    <parse>
        with_priority true
        message_format rfc5424
    </parse>
    tag  pks.default.syslog
</source>

I keep getting this error in the aggregator side (with different applications I deployed  to my PKS cluster - e,g, nginx, wildfly)
2018-10-19 12:31:27 +0000 [warn]: #0 [psksyslog_source] invalid syslog message: "157 <14>1 2018-10-19T12:30:47.614468+00:00 7d551d68-9559-4645-980a-d93c72f7cccb default/pod/wildfly-54bf7c95bd-jxbgl/wildfly - - -   JBoss Bootstrap Environment"

Basically nothing gets parsed.

The only thing I figured was that the additional integer at the beginning would casue the in_syslog plugin to complain.  That is supposing that the message would start with <prio>.

Thanks,
Miklós Vlasa

**Comment by wfernandes** [comment]
@vlasamik Apologies for the delay in response.

I believe you are correct. We prepend the length of the syslog messages because we are transporting over TCP. We use the octet counting format (https://tools.ietf.org/html/rfc6587#section-3.4.1).

I found a relevant issue regarding this in fluentd: https://github.com/fluent/fluentd/issues/1270
Specifically this comment is relevant: https://github.com/fluent/fluentd/issues/1270#issuecomment-253176730

Currently, the fluentd in_syslog doesn't support this form of octet counting delimiter: https://docs.fluentd.org/v0.12/articles/in_syslog#tcp-protocol-and-message-delimiter

I believe this is a valid feature to support the __Non-Transparent-Framing__ form of transport. However, there is a bit of work involved and we will probably work on this in the future.

I'm going to update the title of this issue and keep it open so that our team can track it. However, you could ask for fluentd to support the __Octet Counting__ format.

**Comment by vlasamik** [comment]
Thanks for the update !

Just noticed that the in_syslog plugin coming with Fluentd v1.3.0, supports now the octet counting format ( https://github.com/fluent/fluentd/pull/2147 )
I will try with that fluentd version again.
